test/unit/org/apache/cassandra/tools/SSTableExportTest.java

```diff
@@ -43,12 +62,146 @@ public void testSSTableExport_NoArgs()
+            assertTrue(tool.getStdout(), parsed.get(0).get("partition") != null);
+            assertTrue(tool.getStdout(), parsed.get(0).get("rows") != null);
+            assertTrue(tool.getCleanedStderr(), tool.getCleanedStderr().isEmpty());
+            tool.assertOnExitCode();
+        }
+        assertPostTestEnv();
+    }
+
+    @Test
+    public void testCQLRowArg() throws IOException
+    {
+        try (ToolRunner tool = runner.invokeClassAsTool("org.apache.cassandra.tools.SSTableExport", findOneSSTable("legacy_sstables", "legacy_ma_simple"), "-d"))
+        {
+            assertThat(tool.getStdout(), CoreMatchers.startsWith("[0]"));
+            assertTrue(tool.getCleanedStderr(), tool.getCleanedStderr().isEmpty());
+            tool.assertOnExitCode();
+        }
+        assertPostTestEnv();
+    }
+
+    @Test
+    public void testPKOnlyArg() throws IOException
+    {
+        try (ToolRunner tool = runner.invokeClassAsTool("org.apache.cassandra.tools.SSTableExport", findOneSSTable("legacy_sstables", "legacy_ma_simple"), "-e"))
+        {
+            assertEquals(tool.getStdout(), "[ [ \"0\" ], [ \"1\" ], [ \"2\" ], [ \"3\" ], [ \"4\" ]\n]", tool.getStdout());
+            assertTrue(tool.getCleanedStderr(), tool.getCleanedStderr().isEmpty());
+            tool.assertOnExitCode();
+        }
+        assertPostTestEnv();
+    }
+
+    @Test
+    public void testPKArg() throws IOException
+    {
+        try (ToolRunner tool = runner.invokeClassAsTool("org.apache.cassandra.tools.SSTableExport", findOneSSTable("legacy_sstables", "legacy_ma_simple"), "-k", "0"))
+        {
+            List<Map<String, Object>> parsed = mapper.readValue(tool.getStdout(), jacksonListOfMapsType);
+            assertEquals(tool.getStdout(), 1, parsed.size());
+            assertEquals(tool.getStdout(), "0", ((List) ((Map) parsed.get(0).get("partition")).get("key")).get(0));
+            assertTrue(tool.getCleanedStderr(), tool.getCleanedStderr().isEmpty());
+            tool.assertOnExitCode();
+        }
+        assertPostTestEnv();
+    }
+
+    @Test
+    public void testExcludePKArg() throws IOException
+    {
+        try (ToolRunner tool = runner.invokeClassAsTool("org.apache.cassandra.tools.SSTableExport", findOneSSTable("legacy_sstables", "legacy_ma_simple"), "-x", "0"))
+        {
+            List<Map<String, Object>> parsed = mapper.readValue(tool.getStdout(), jacksonListOfMapsType);
+            assertEquals(tool.getStdout(), 4, parsed.size());
+            assertTrue(tool.getCleanedStderr(), tool.getCleanedStderr().isEmpty());
+            tool.assertOnExitCode();
+        }
+        assertPostTestEnv();
+    }
+
+    @Test
+    public void testTSFormatArg() throws IOException
+    {
+        try (ToolRunner tool = runner.invokeClassAsTool("org.apache.cassandra.tools.SSTableExport", findOneSSTable("legacy_sstables", "legacy_ma_simple"), "-t"))
+        {
+            assertThat(tool.getStdout(),
```

Comment:
this should check the actual field has the correct value rather than just contains.